src/adera/Machines/Rocket.h

```diff
@@ -81,13 +81,21 @@ class SysMachineRocket :
 class MachineRocket : public osp::active::Machine
 {
     friend SysMachineRocket;
+
+    using input_t = std::pair<osp::DependRes<ShipResourceType>, float>;
     struct ResourceInput
     {
+        ResourceInput(osp::DependRes<ShipResourceType> type,
+            float massRateFraction, osp::active::WireInput&& source)
```

Comment:
generally i discourage taking a parameter by &&, preferring to take by value or const& only, unless i have two versions, one for const& and one for &&.

If you want to deviate from that, I recommend providing a comment elaborating on why.